halo/attest/keeper/keeper_test.go

```diff
@@ -0,0 +1,69 @@
+package keeper_test
+
+import (
+	"testing"
+
+	"github.com/omni-network/omni/halo/attest/keeper"
+	"github.com/omni-network/omni/halo/attest/types"
+
+	k1 "github.com/cometbft/cometbft/crypto/secp256k1"
+	cmttypes "github.com/cometbft/cometbft/types"
+
+	ormv1alpha1 "cosmossdk.io/api/cosmos/orm/v1alpha1"
+	"cosmossdk.io/orm/model/ormdb"
+	fuzz "github.com/google/gofuzz"
+	"github.com/stretchr/testify/require"
+)
+
+func TestKeeper_Add(t *testing.T) {
+	t.Parallel()
+
+	k, store, ctx := setupKeeper(t)
+
+	var votes []*types.AggVote
+	fuzz.New().NilChance(0).NumElements(1, 3).Fuzz(&votes)
+	msg := &types.MsgAddVotes{
+		Authority: "test-authority",
+		Votes:     votes,
+	}
+
+	err := k.Add(ctx, msg)
+	require.NoError(t, err)
+
+	schema := &ormv1alpha1.ModuleSchemaDescriptor{SchemaFile: []*ormv1alpha1.ModuleSchemaDescriptor_FileEntry{
```

Comment:
not sure constructing all this again outside the keeper is correct. Why not just provide a getter for the table on the keeper? Can do something like;
```
// AttTableForT returns the attestestation table for testing purposes. 
func (k Keeper) AttTableForT(*testint.T) AttestationTable() {
```

Reply:
Other option is to return the `attstore.AttestationTable` from `setupKeeper` so each test doesn't need to setup all this manually.